### ISSUE TITLE
Settings UI should use list accessibilityProvider

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#95862]
Curretnly the settings UI uses a custom accessibility solution. Code pointer https://github.com/microsoft/vscode/blob/07bd3fff5c576e539f3079c09a46d15d5af98338/src/vs/workbench/contrib/preferences/browser/settingsTree.ts#L554

Instead the settings tree should use the AccessibilityProvider provided by the tree and if something is missing this should be added as a list/tree aria feature.

fyi @joaomoreno

**Comment by joaomoreno** [comment]
@roblourens Do let me know if you need more a11y features in the list/tree, instead of manipulating the DOM directly.

**Comment by isidorn** [comment]
Let's assign this to next milestone and me so we can look into it together @roblourens

**Comment by roblourens** [comment]
I don't know if that makes sense, the list AccessibilityProvider sets aria properties on the list rows themselves, but the settings tree needs to set them on elements inside each row which are owned by the renderer. I don't know how the tree would do that and I don't think it makes sense.

**Comment by isidorn** [comment]
@roblourens ok, but why does the settings tree need to set them on elements inside each row?
If you prefer we can have a call where you can give me more details about how accessibility is working in the settings UI and we can toss ideas around on how to potentially fix this. Let me know what you think.

**Comment by roblourens** [comment]
Because that's what gets focused. The settings tree is an oddball list in that the list rows don't get focused, the elements inside do and you navigate it like a form, not a typical list.

**Comment by isidorn** [comment]
Yeah I have always found that strange. And I think quite some screen reader users find that strange as well.
These two issues seem to caputre that well
https://github.com/microsoft/vscode/issues/92488
https://github.com/microsoft/vscode/issues/92485

If we have plans to fix that in the future maybe then is the right time to look into this issue as well.
IMHO settings navigation should work like any other lists, since our users are accustomed the tree navigation.
So up / down goes through each element and space or tab move focus inside the elment to change the value.

**Comment by roblourens** [comment]
Yeah I know it's come up but if we're going to redesign it to assist screenreader users, there are other issues to resolve like the extremely long flat virtualized list.

**Comment by roblourens** [comment]
I'm gonna close this as the general issue is covered other places like you pointed out.
